### PR TITLE
Support generating external adapters

### LINKED ISSUE #119: Support generation of adapters outside of autovalue class hierarchy
Matching auto-value-gson's support for this. Requires a custom factory as well

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #119 

This uses moshi's native generated adapter lookup mechanism. To use, simply annotate the autovalue class with `@JsonClass(generateAdapter = true, generator = "avg")` and then auto-value-moshi will generate a top-level adapter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,9 +6,37 @@ An extension for Google's [AutoValue](https://github.com/google/auto) that creat
 
 ## Usage
 
-Simply include auto-value-moshi in your project and add a public static method with the following
-signature to classes you want to get Moshi `JsonAdapter`s. You can also annotate your properties
-using `@Json` to define an alternate name for de/serialization.
+Simply include auto-value-moshi in your project and annotate your target autovalue class with Moshi's
+`@JsonClass` annotation. `generateAdpater` must be true, and the `generator` property value should
+be `"avg"`.
```
Comment (rharter) [inline README.md]: Should be "autovalue", or at least "avm", right?

Comment (ZacSweers) [reply README.md]: d'oh, fixed avm

**Review by rharter — APPROVED** [review]
Very nice.

This gets me wondering if there's a way for the `generator` property to be inferred, like a service locator in Moshi that effectively tries to find services that can generate the adapter, in this case AVM providing a generator for classes that are `@JsonClass` and `@AutoValue` annotated, otherwise falling back to it's own internal generator. That is most definitely a nice-to-have that's out of this scope, though, and possible not even worth it.

**Comment by ZacSweers** [comment]
Possibly! R8 does support inlining servicelocator lookups so it's something we could look at. Will file an issue

**Pull request merged**